src/test/java/com/google/sps/servlets/UserServletTest.java

```diff
@@ -0,0 +1,197 @@
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import java.io.Reader;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+public class UserServletTest {
+
+    private Entity user1;
+    private Entity user2;
+    private Entity user3;
+
+    private final LocalServiceTestHelper helper =
+        new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void testDoGet() throws IOException, ServletException {
+
+        DatastoreService localDatabase = DatastoreServiceFactory.getDatastoreService();
+
+        caller = new Entity("user");
+        caller.setProperty("userId", "11234567890");
+        caller.setProperty("name", "caller");
+        caller.setProperty("email", "caller@google.com");
+        caller.setProperty("language", "english");
+
+        user1 = new Entity("user");
+        user1.setProperty("userId", "21234567890");
+        user1.setProperty("name", "user1");
+        user1.setProperty("email", "user1@google.com");
+        user1.setProperty("language", "spanish");
+
+        user2 = new Entity("user");
+        user2.setProperty("userId", "31234567890");
+        user2.setProperty("name", "user2");
+        user2.setProperty("email", "user2@google.com");
+        user2.setProperty("language", "chinese");
+
+        localDatabase.put(caller);
+        localDatabase.put(user1);
+        localDatabase.put(user2);
+
+        UserServlet userServlet = new UserServlet();
+        HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+        HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        PrintWriter printWriter = Mockito.mock(PrintWriter.class);
+
+        Mockito.when(request.getParameter("userId").thenReturn((String) caller.getProperty("userId")));
+        Mockito.when(response.getWriter()).thenReturn(printWriter);
+
+        userServlet.doGet(request, response);
+
+        ArrayList<User> users = new ArrayList<User>();
+
+        User user = new User((String) caller.getProperty("userId"),
+        (String) caller.getProperty("name"),
+        (String) caller.getProperty("email"),
+        (String) caller.getProperty("language"));
+
+        users.add(user);
+        Mockito.verify(response).setContentType("application/json");
+        Mockito.verify(printWriter).println(gson.toJson(users));
+    }
+
+    @Test
```

Comment:
Please add comment of how the mockito testing works here

Reply:
Will do.